internal/cmd/perfcomp/main.go

```diff
@@ -0,0 +1,270 @@
+	}
+
+	if err := cursor.Err(); err != nil {
+		return nil, err
+	}
+
+	return rawData, nil
+}
+
+// Find the most recent stable region of the mainline version for a specific test/measurement
+func findLastStableRegion(testname string, measurement string, coll *mongo.Collection) (*StableRegion, error) {
+	filter := bson.D{
+		{"time_series_info.project", "mongo-go-driver"},
+		{"time_series_info.variant", "perf"},
+		{"time_series_info.task", "perf"},
+		{"time_series_info.test", testname},
+		{"time_series_info.measurement", measurement},
+		{"last", true},
+		{"contexts", []string{"GoDriver perf (h-score)"}},
+	}
+
+	findOptions := options.FindOne().SetSort(bson.D{{"end", -1}})
+
+	findCtx, cancel := context.WithTimeout(context.Background(), 180*time.Second)
+	defer cancel()
+
+	var sr *StableRegion
+	err := coll.FindOne(findCtx, filter, findOptions).Decode(&sr)
+	if err != nil {
+		return nil, err
+	}
+	return sr, nil
+}
+
+// For a specific test and measurement
+func getEnergyStatsForOneBenchmark(rd RawData, coll *mongo.Collection) ([]*EnergyStats, error) {
+	testname := rd.Info.TestName
+	var energyStats []*EnergyStats
+
+	for i := range rd.Rollups.Stats {
+		measurement := rd.Rollups.Stats[i].Name
+		patchVal := []float64{rd.Rollups.Stats[i].Val}
+
+		stableRegion, err := findLastStableRegion(testname, measurement, coll)
+		if err != nil {
+			return nil, err
+		}
+
+		pChange := GetPercentageChange(patchVal[0], stableRegion.Mean)
+		e, t, h := GetEnergyStatistics(mat.NewDense(len(stableRegion.Values), 1, stableRegion.Values), mat.NewDense(1, 1, patchVal))
+		z := GetZScore(patchVal[0], stableRegion.Mean, stableRegion.Std)
+
+		es := EnergyStats{
+			Benchmark:    testname,
+			Measurement:  measurement,
+			PatchVersion: rd.Info.Version,
+			StableRegion: *stableRegion,
+			PatchValues:  patchVal,
+			PChange:      pChange,
+			E:            e,
+			T:            t,
+			H:            h,
+			Z:            z,
+		}
+		energyStats = append(energyStats, &es)
+	}
+
+	return energyStats, nil
+}
+
+func getEnergyStatsForAllBenchMarks(patchRawData []RawData, coll *mongo.Collection) ([]*EnergyStats, error) {
+	var allEnergyStats []*EnergyStats
+	for _, rd := range patchRawData {
+		energyStats, err := getEnergyStatsForOneBenchmark(rd, coll)
+		if err != nil {
+			return nil, err
+		}
+		allEnergyStats = append(allEnergyStats, energyStats...)
+	}
+	return allEnergyStats, nil
+}
+
+func generatePRComment(energyStats []*EnergyStats, version string) string {
+	var comment strings.Builder
+	comment.WriteString("# 👋GoDriver Performance\n")
+	fmt.Fprintf(&comment, "The following benchmark tests for version %s had statistically significant changes (i.e., |z-score| > 1.96):\n", version)
+	comment.WriteString("| Benchmark | Measurement | H-Score | Z-Score | % Change | Stable Reg | Patch Value |\n| --- | --- | --- | --- | --- | --- | --- |\n")
+
+	var testCount int64
+	for _, es := range energyStats {
+		if es.Z > 1.96 || es.Z < -1.96 {
```

Comment:
Are the positive and negative range always the same magnitude? If so, you can use `math.Abs`.


```suggestion
		if math.Abs(es.Z) > 1.96 {
```